lib/rouge/lexers/lustre.rb

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*- #
+#
+# adapted from ocaml.rb, hence some ocaml-ism migth remains
+module Rouge
+  module Lexers
+    class Lustre < RegexLexer
+      title "Lustre"
+      desc 'The Lustre programming language (Verimag)'
+      tag 'lustre'
+      filenames '*.lus'
+      mimetypes 'text/x-lustre'
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          extern unsafe assert const current enum function
+          false let node operator returns
+          step struct tel type true var  model package needs
+          provides uses is body end include merge
+        )
+      end
+
+      def self.word_operators
+        @word_operators ||= Set.new %w(
+           div and xor mod or not nor if then else fby pre when with) 
+      end
+
+      def self.primitives
+        @primitives ||= Set.new %w(int real bool)
+      end
+
+      operator = %r([;,_!$%&*+./:<=>?@^|~#-]+)
+      id = /[a-z_][\w']*/i
+
+      state :root do
+        rule /\s+/m, Text
+        rule /false|true|[(][)]|\[\]/, Name::Builtin::Pseudo
+        rule %r(\-\-.*?$), Comment::Single
+        rule %r(/\*.*?\*/)m, Comment::Multiline
+        rule %r(\(\*.*?\*\))m, Comment::Multiline
+        rule id do |m|
+          match = m[0]
+          if self.class.keywords.include? match
+            token Keyword
+          elsif self.class.word_operators.include? match
+            token Operator::Word
+          elsif self.class.primitives.include? match
+            token Keyword::Type
+          else
+            token Name
+          end
+        end
+
+        rule /[(){}\[\];]+/, Punctuation
```

Comment:
You have as both punctuation and as an operator. The operator token will never match. Is that OK?

Reply:
I will remove the line

Reply:
Actually removing that line broke test. I've put it back.
Could you pleasereword the sentence "You have as both punctuation and as an operator" ?
I don't understand it.

Reply:
Sorry—the initial comment was poorly worded >_<

What I was trying to say was that this line and the line that comes after it (`rule operator, Operator`) both match against `;`. The comment meant to ask: how should `;` be lexed? Tokenising it as `Punctuation` looks correct to me but if that's the case, it should be removed from the definition of `operator`.

Speaking of which, should some of the other characters be moved out of the `operator` rule? Is `,` or `_` really an operator?

Reply:
ok, thnaks for the clarification. I'll remove ';' from the definition of operator.

Concerning ',' and '_',  maybe not. Once again, I've taken inspiration from ocaml.rb, from which the role of ',' '_', or ';' is similar than in Lustre to some extent. In Lustre clearly, ';' is a punctuation (in ocaml too). ',' appears in structures and arrays, and it may make sense to see them as operator. '_' is an identifier as in ocaml; i'll remove it from the operator section.

Reply:
There's some value in being consistent with previous lexers and if Lustre works similarly to OCaml then maintaining consistency makes sense. I'm not very familiar with OCaml but it does seem that `;` and `,` are referred to as operators in documentation ([example](https://ocaml.org/learn/tutorials/structure_of_ocaml_programs.html)) so if it's performing a similar function in Lustre, putting it back seems fine to me.

Reply:
Well, I'am not sure seing ";" as an operator in Lustre makes sense as it does in ocaml. Therefore I propose to leave it as it is currently.